### PR TITLE
GIT-4698: enable clustermgr base UTs

### LINKED ISSUE #4698: DDE: Add unit test coverage
## Feature Request

This is to track adding unit tests to cover the Docker Desktop Extension code.

The expectation is not to have 100% code coverage, nor any predetermined coverage level. The code should be reviewed to determine what is testable - or can be made testable - and close out this issue when it's decided that a reasonable level has been reached.

### PR DESCRIPTION
**Pull request by harshanarayana**
## What this PR does / why we need it
Add Additional UT coverage for DDE Extension

## Which issue(s) this PR fixes
Fixes: #4698

## Describe testing done for PR
UT Coverage ran locally to confirm no builds were broken and tests 

## Special notes for your reviewer
There is minor refactoring done to the `docker.go` to enable better re-factoring. 

This PR introduces a stubbed out interface named `DockerInterface` which is a watered down version of the functions exposed by the `client.Client` struct to enable better mocking and testability of the DDE helpers.

### CODE REVIEW

File: extensions/docker-desktop/apps/clustermgr/pkg/internal/docker/docker_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package docker
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/mock"
+	"github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/internal/docker/mocks"
```
Comment (stmcginnis) [inline extensions/docker-desktop/apps/clustermgr/pkg/internal/docker/docker_test.go:13]: Will need a line break here to separate the third party imports from this local import to make the linter happy.

File: extensions/docker-desktop/apps/clustermgr/pkg/kubeconfig/internal/kubeconfig/main_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeconfig
+
+import (
+	"io/ioutil"
```
Comment (stmcginnis) [inline extensions/docker-desktop/apps/clustermgr/pkg/kubeconfig/internal/kubeconfig/main_test.go:7]: `ioutil` should not be used as it was deprecated in go 1.16.

**Review by stmcginnis — COMMENTED** [review]
This is looking great - thanks for working on it!

File: extensions/docker-desktop/apps/clustermgr/pkg/kubeconfig/internal/kubeconfig/main_test.go (hunk 2 above)
Comment (harshanarayana) [reply extensions/docker-desktop/apps/clustermgr/pkg/kubeconfig/internal/kubeconfig/main_test.go:7]: 🤦 Fixing it right-away.

File: extensions/docker-desktop/apps/clustermgr/pkg/internal/docker/docker_test.go (hunk 1 above)
Comment (harshanarayana) [reply extensions/docker-desktop/apps/clustermgr/pkg/internal/docker/docker_test.go:13]: /Ack

**Comment by stmcginnis** [comment]
Thanks for the quick update! Should this still be marked as a draft?

**Comment by harshanarayana** [comment]
@stmcginnis Not anymore. Moving this out of Draft state. Also added a few more tests for the `read.go`

**Comment by stmcginnis** [comment]
Looks like you'll need to run `make get-deps` to get updates for `go.mod` and `go.sum`.

**Comment by harshanarayana** [comment]
> Looks like you'll need to run `make get-deps` to get updates for `go.mod` and `go.sum`.

@stmcginnis  Updated. PTAL.

**Review by stmcginnis — APPROVED** [review]
Awesome, from zero test coverage to fairly decent.

Before:

```
❯ make test
go test ./... -coverprofile=cover.out
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/checks	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/cluster	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/config	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/internal/docker	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig/internal/kubeconfig	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig/internal/sets	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/utils	[no test files]
```

No test coverage. Then after:

```
❯ gh co 4762
...
❯ make test
go test ./... -coverprofile=cover.out
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop	[no test files]
ok  	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/checks	0.297s	coverage: 72.4% of statements
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/cluster	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/config	[no test files]
ok  	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/internal/docker	0.439s	coverage: 57.1% of statements
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/internal/docker/mocks	[no test files]
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig	[no test files]
ok  	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig/internal/kubeconfig	0.290s	coverage: 61.0% of statements
ok  	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/kubeconfig/internal/sets	0.472s	coverage: 87.1% of statements
?   	github.com/vmware-tanzu/community-edition/extensions/docker-desktop/pkg/utils	[no test files]
```

**Pull request merged**